src/pdfkit.cs

```diff
@@ -86,53 +495,74 @@ interface PdfActionRemoteGoTo {
 		NSUrl Url { get; set; }
 	}
 
-	[BaseType (typeof (PdfAction), Name="PDFActionResetForm")]
+	[iOS (11,0)]
+	[BaseType (typeof (PdfAction), Name = "PDFActionResetForm")]
 	interface PdfActionResetForm {
 		//has a public Init ???
-		
+
 		//NSArray of NSString
 		[Export ("fields"), NullAllowed]
 		string [] Fields { get; set; }
 
 		[Export ("fieldsIncludedAreCleared")]
-		bool FieldsIncludedAreCleared  { get; set; }
+		bool FieldsIncludedAreCleared { get; set; }
 	}
 
-	[BaseType (typeof (PdfAction), Name="PDFActionURL")]
+	[iOS (11,0)]
+	[BaseType (typeof (PdfAction), Name = "PDFActionURL")]
 	interface PdfActionUrl {
+
+		[DesignatedInitializer]
 		[Export ("initWithURL:")]
 		IntPtr Constructor (NSUrl url);
 
 		[Export ("URL")]
 		NSUrl Url { get; set; }
 	}
 
-	[BaseType (typeof (NSObject), Name="PDFAnnotation")]
+	[iOS (11,0)]
+	[BaseType (typeof (NSObject), Name = "PDFAnnotation")]
 	interface PdfAnnotation : NSCoding, NSCopying {
+
+		[Mac (10,13)]
+		[Export ("initWithBounds:forType:withProperties:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (CGRect bounds, NSString annotationType, [NullAllowed] NSDictionary properties);
+
+		[Mac (10,13)]
+		[Wrap ("this (bounds, annotationType.GetConstant (), properties)")]
+		IntPtr Constructor (CGRect bounds, PDFAnnotationKey annotationType, [NullAllowed] NSDictionary properties);
+
+		[Deprecated (PlatformName.iOS, 11, 0, message: "Use '.ctor (CGRect, PDFAnnotationKey, NSDictionary)' instead")]
+		[Deprecated (PlatformName.MacOSX, 10, 13, message: "Use '.ctor (CGRect, PDFAnnotationKey, NSDictionary)' instead")]
 		[Export ("initWithBounds:")]
 		IntPtr Constructor (CGRect bounds);
 
 		[Export ("page")]
-		PdfPage Page { get; }
+		PdfPage Page { get; set; }
 
 		[Export ("type")]
-		string Type { get; }
+		string Type { get; set; }
+
+		[Wrap ("PDFAnnotationKeyExtensions.GetValue ((NSString) Type)")]
+		PDFAnnotationKey AnnotationType { get; }
 
 		[Export ("bounds")]
 		CGRect Bounds { get; set; }
-		
+
```

Comment:
Even if I agree the tabs shouldn't be there, it's still noise, we should avoid such change to ease merging.